spot-contracts/contracts/FeePolicy.sol

```diff
@@ -172,205 +154,97 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
-    /// @param debasementProtocolSharePerc_ The share of the debasement which goes to the protocol.
-    /// @param enrichmentProtocolSharePerc_ The share of the enrichment which goes to the protocol.
-    function updateProtocolSharePerc(
-        uint256 debasementProtocolSharePerc_,
-        uint256 enrichmentProtocolSharePerc_
-    ) external onlyOwner {
-        if (debasementProtocolSharePerc_ > ONE || enrichmentProtocolSharePerc_ > ONE) {
+    /// @notice Updates the protocol share of tvl extracted on every rebalance.
+    /// @param protocolSharePerc_ The share of the tvl which goes to the protocol as a percentage.
+    /// @param protocolFeeCollector_ The new fee collector address.
+    function updateProtocolFeeConfig(uint256 protocolSharePerc_, address protocolFeeCollector_) external onlyOwner {
+        if (protocolSharePerc_ > ONE) {
             revert InvalidPerc();
         }
-
-        debasementProtocolSharePerc = debasementProtocolSharePerc_;
-        enrichmentProtocolSharePerc = enrichmentProtocolSharePerc_;
+        protocolSharePerc = protocolSharePerc_;
+        protocolFeeCollector = protocolFeeCollector_;
     }
 
     //-----------------------------------------------------------------------------
     // Public methods
 
     /// @inheritdoc IFeePolicy
-    /// @dev Minting perps reduces system dr, i.e) drPost < drPre.
-    function computePerpMintFeePerc() public view override returns (uint256) {
-        return perpMintFeePerc;
-    }
-
-    /// @inheritdoc IFeePolicy
-    /// @dev Burning perps increases system dr, i.e) drPost > drPre.
-    function computePerpBurnFeePerc() public view override returns (uint256) {
-        return perpBurnFeePerc;
-    }
-
-    /// @inheritdoc IFeePolicy
-    /// @dev Minting vault notes increases system dr, i.e) drPost > drPre.
-    function computeVaultMintFeePerc() external view override returns (uint256) {
-        return vaultMintFeePerc;
-    }
-
-    /// @inheritdoc IFeePolicy
-    function computeVaultBurnFeePerc() external view override returns (uint256) {
-        return vaultBurnFeePerc;
-    }
-
-    /// @inheritdoc IFeePolicy
-    /// @dev Swapping by minting perps reduces system dr, i.e) drPost < drPre.
-    function computeUnderlyingToPerpVaultSwapFeePerc(
-        uint256 drPre,
-        uint256 drPost
-    ) external view override returns (uint256) {
-        // When the after op deviation ratio is below the bound,
-        // swapping is disabled. (fees are set to 100%)
-        if (drPost <= drHardBound.lower) {
-            return ONE;
+    function computeFeePerc(uint256 drPre, uint256 drPost) public view override returns (uint256) {
+        // DR is decreasing, we use feeFnDRDown
+        if (drPre > drPost) {
+            Line memory fee = feeFnDRDown;
+            return
+                LineHelpers
+                    .computePiecewiseAvgY(
+                        fee,
+                        Line({ x1: ONE, y1: fee.y2, x2: fee.x2, y2: fee.y2 }),
```

Comment:
I expected it to go like:
`Line({ x1: fee.x2, y1: fee.y2, x2: ONE, y2: fee.y2 }),`

So we keep the pattern of going left to right